### ISSUE TITLE
how to use vue-devtools in chrome extension options page

### ISSUE THREAD

**Issue by honwhy** [vuejs/devtools#727]
failed to load virtual js files from vue-devtools when running in chrome extension page.
how to specify injected js src with origin from dev server.

I am using wxt.dev to develop chrome extension, wxt.config.ts config like this,

```
// wxt.config.ts
import vueDevTools from 'vite-plugin-vue-devtools'

export default defineConfig({
  plugins: [
   vue(),
   vueDevtools()
  ]
})
```
something i found after start `npx wxt` on my project,
```
//.output/chrome-mv3/options.html
<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <script type="module" src="/@id/virtual:vue-devtools-path:overlay.js"></script>
    <script type="module" src="/@id/virtual:vue-inspector-path:load.js"></script>
....
```
as a result, request is failed
```
Request URL:
chrome-extension://amidodbgcdpbldlmccigekeemjhkkhlh/@id/virtual:vue-devtools-path:overlay.js
Referrer Policy:
strict-origin-when-cross-origin
```

it would be correct if the injected js src to be prepend with dev server,  for example
```
    <script type="module" src="http://localhost:3000/@id/virtual:vue-devtools-path:overlay.js"></script>
    <script type="module" src="http://localhost:3000/@id/virtual:vue-inspector-path:load.js"></script>
```

**Comment by alexzhang1030** [comment]
If you are using `WXT`, try this

https://devtools.vuejs.org/help/faq#how-to-work-with-wxt

**Comment by honwhy** [comment]
Yeah, I am using `WXT`, work around solution may like the adding another plugin,
```
export function vueDevtoolsHack(
  config: wxt.ResolvedConfig,
  getWxtDevServer: () => wxt.WxtDevServer | undefined,
): vite.Plugin {
  const server = getWxtDevServer?.()
  return {
    name: `md:vue-devtools-hack`,
    apply: `build`,
    transformIndexHtml: {
      order: `post`,
      handler(html) {
        const { document } = parseHTML(html)
        const inlineScripts = document.querySelectorAll(`script[src^='/@id/virtual:']`)
        inlineScripts.forEach((script) => {
          const src = script.getAttribute(`src`)
          const newSrc = `${server?.origin}${src}`
          script.setAttribute(`src`, newSrc)
        })
        const newHtml = document.toString()
        config.logger.debug(`Old HTML:\n${html}`)
        config.logger.debug(`New HTML:\n${newHtml}`)
        return newHtml
      },
    },
  }
}

```

**Comment by alexzhang1030** [comment]
We have supported this by configuring `appendTo`, which works out-of-box by passing to vite. If you have some other special cases, let's discuss them.

**Comment by honwhy** [comment]
Note that there is a slight difference in my setup: the entry point is dynamically added through a WXT hook.
https://github.com/doocs/md/blob/79477702d245b66ce1451ec99227d16e2f729e46/src/modules/build-extension.ts#L20

**Comment by alexzhang1030** [comment]
This is your entry point I guess. Can you try it out?

![image](https://github.com/user-attachments/assets/d7e11dcb-cd06-4b1e-95ad-ecce4e59d7fa)

**Comment by DES2048** [comment]
Ufortunately setting appendTo: "/entrypoints/options/main.ts" doesnt working for me. Can you explain how to properly set this option? I tried to set appendTo starting with src: "/src/entrypoints/options/main.ts" and with full url: http://localhost:3000/src/entrypoints/options/main.ts". But it still doesnt work:(. But in rendered main.ts I saw that devtools imports is present:
`import "/node_modules/.pnpm/vite-plugin-vue-devtools@7.6.8_rollup@4.28.1_vite@5.4.11_@types+node@22.10.1__vue@3.5.13_typescript@5.6.3_/node_modules/vite-plugin-vue-devtools/src/overlay.js?__vue-devtools-next-resource";
import { createApp } from "/node_modules/.vite/deps/vue.js?v=1f931d2c";
import vuetify from "/src/plugins/vuetify.ts";
import Options2 from "/src/entrypoints/options/Options2.vue";
const app = createApp(Options2).use(vuetify);
app.mount("#app");
import "/node_modules/.pnpm/vite-plugin-vue-inspector@5.3.1_vite@5.4.11_@types+node@22.10.1_/node_modules/vite-plugin-vue-inspector/src/load.js";
`

**Comment by marijnbent** [comment]
The appendTo doesn't work for me either
